src/Uno.UI/UI/Xaml/Controls/Video/HtmlMediaPlayer.wasm.cs

```diff
@@ -0,0 +1,482 @@
+					{
+						_htmlVideo.RemoveAttribute("controls");
+					}
+					else
+					{
+						_htmlVideo.SetAttribute("controls", "controls");
+					}
+				}
+				else if (player.IsAudio)
+				{
+					if (!string.IsNullOrEmpty(_htmlAudio.GetAttribute("controls")))
+					{
+						_htmlAudio.RemoveAttribute("controls");
+
+					}
+					else
+					{
+						_htmlAudio.SetAttribute("controls", "controls");
+					}
+				}
+			}
+			else
+			{
+				if (player.IsVideo)
+				{
+					if (!string.IsNullOrEmpty(_htmlVideo.GetAttribute("controls")))
+					{
+						_htmlVideo.RemoveAttribute("controls");
+					}
+				}
+				else if (player.IsAudio)
+				{
+					if (!string.IsNullOrEmpty(_htmlAudio.GetAttribute("controls")))
+					{
+						_htmlAudio.RemoveAttribute("controls");
+					}
+				}
+			}
+		}
+	}
+
+	public string Source
+	{
+		get => (string)GetValue(SourceProperty);
+		set => SetValue(SourceProperty, value);
+	}
+
+	public bool AutoPlay
+	{
+		get => (bool)GetValue(AutoPlayProperty);
+		set => SetValue(AutoPlayProperty, value);
+	}
+
+	public bool AreTransportControlsEnabled
+	{
+		get => (bool)GetValue(AreTransportControlsEnabledProperty);
+		set => SetValue(AreTransportControlsEnabledProperty, value);
+	}
+
+	public void RequestFullScreen()
+	{
+		var js = $"var elem = Uno.UI.WindowManager.current.getView({_htmlVideo.HtmlId});" +
+				"if (elem.requestFullscreen) {elem.requestFullscreen();} else if (elem.mozRequestFullScreen) {elem.mozRequestFullScreen();} else if (elem.webkitRequestFullscreen) {elem.webkitRequestFullscreen();} else if (elem.msRequestFullscreen) {elem.msRequestFullscreen();}";
+		WebAssemblyRuntime.InvokeJS(js);
+	}
+
+	public void ExitFullScreen()
+	{
+		var js = "if (document.exitFullscreen) {document.exitFullscreen();} else if (document.msExitFullscreen) {document.msExitFullscreen();} else if (document.mozCancelFullScreen) {document.mozCancelFullScreen();} else if (document.webkitExitFullscreen) {document.webkitExitFullscreen();}";
+		WebAssemblyRuntime.InvokeJS(js);
+
+	}
+
+	public void Pause()
+	{
+		var js = $"var element = Uno.UI.WindowManager.current.getView({(IsAudio ? _htmlAudio.HtmlId : _htmlVideo.HtmlId)});" +
+				$"element.pause();";
+		WebAssemblyRuntime.InvokeJS(js);
+	}
+
+	public void Play()
+	{
+		var js = $"var element = Uno.UI.WindowManager.current.getView({(IsAudio ? _htmlAudio.HtmlId : _htmlVideo.HtmlId)});" +
+				$"element.play();";
+		WebAssemblyRuntime.InvokeJS(js);
+	}
+
+	public void Stop()
+	{
+		var js = $"var element = Uno.UI.WindowManager.current.getView({(IsAudio ? _htmlAudio.HtmlId : _htmlVideo.HtmlId)});" +
+				$"element.pause();" +
+				$"element.currentTime = 0;";
+		WebAssemblyRuntime.InvokeJS(js);
+	}
+
+	public void ConfigureMetadata(bool isAudio)
+	{
+		var js = $"var element = Uno.UI.WindowManager.current.getView({(isAudio ? _htmlAudio.HtmlId : _htmlVideo.HtmlId)});" +
+				$"element.preload = 'metadata';";
+		WebAssemblyRuntime.InvokeJS(js);
+	}
+
+	public new void Dispose()
```

Comment:
hiding is quite weird here. Where base implementation come from? I can't guess what in the inheritance hierarchy is disposable

Reply:
That will be FrameworkElement